cpp/src/arrow/csv/reader.h

```diff
@@ -101,6 +102,10 @@ class ARROW_EXPORT StreamingReader : public RecordBatchReader {
       io::IOContext io_context, std::shared_ptr<io::InputStream> input,
       arrow::internal::Executor* cpu_executor, const ReadOptions&, const ParseOptions&,
       const ConvertOptions&);
+  static Future<std::shared_ptr<StreamingReader>> MakeAsync(
```

Comment:
Let's add at least a brief comment here explaining the difference between the two reader types.